### DISCUSSION TITLE
ESP-Now Documentation

### DISCUSSION THREAD

**Discussion by neliel123** [micropython/micropython#12353] [ESP32]
I am a newbie in MicroPython and I wanted to explore the ESP-NOW implementation in MicroPython. I only see the following documentation https://docs.micropython.org/en/latest/library/espnow.html

While trying to Google, a lot of post references this https://micropython-glenn20.readthedocs.io/ and looks like this is where the fork for ESP-NOW MicroPython implementation was created.  It is returning a 404 right now though.

I am just wondering if there is a backup for this. Looks like there are a lot of good stuff to read there.

For a beginner, there are not many tutorials out there per see.

I was actually looking for some examples to get started

**Comment by PatLub85** [comment]
Hi,
have you looked at these examples?
https://github.com/glenn20/micropython-espnow-utils

**Comment by shariltumin** [comment]
Personally, I think the [documentation](https://docs.micropython.org/en/latest/library/espnow.html) is pretty good. The [test scripts](https://github.com/micropython/micropython/tree/master/tests/multi_espnow) also give some nice examples.

You might be interested to see a practical use case of ESP-NOW for [dynamic-mesh](https://github.com/shariltumin/mesh-espnow-micropython). The work is based on the [worker-framework](https://github.com/shariltumin/workers-framework-micropython), which in my opinion is easier than asyncio. You are invited to have a look. You might find it useful.

**Comment by glenn20** [comment]
There was no additonal information available at https://micropython-glenn20.readthedocs.io/en/latest/library/espnow.html, that is not included in https://docs.micropython.org/en/latest/library/espnow.html (as far as I remember).

<https://micropython-glenn20.readthedocs.io/en/latest/library/espnow.html> held the docs for the `espnow` module as they evolved over the last few years (I just needed somewhere to host the docs prior to the merge). Those docs were all merged into micropython along with the `espnow` module code ([PR 6515](https://github.com/micropython/micropython/pull/6515)) and are now available at https://docs.micropython.org/en/latest/library/espnow.html. I took down the former as the micropython docs are where this belongs now and there should really be just one source of truth for the docs.

**Comment by glenn20** [comment]
> For a beginner, there are not many tutorials out there per see.
> 
> I was actually looking for some examples to get started

This is a good point. I will look at wrapping up some of my own code samples and making them available as documented introductory examples. I tried to provide plenty of example code snippets in the docs, but a few more comprehensive (yet introductory) application examples in a repo might be helpful.

**Comment by MrTinkerman** [comment]
I also just started exploring esp-now and found this:
https://www.donskytech.com/tag/espnow/
There's also a github and youtube by DonskyTech

I'm not a complete beginner but I think these are pretty basic and understandable examples.
Be sure to get a grasp of asyncio though.
